src/main/java/org/jboss/ejb/protocol/remote/RemotingEJBClientInterceptor.java

```diff
@@ -74,13 +75,19 @@ public SessionID handleSessionCreation(final EJBSessionCreationInvocationContext
     private void removeNode(final AbstractInvocationContext context) {
         final Affinity targetAffinity = context.getTargetAffinity();
         if (targetAffinity instanceof NodeAffinity) {
-            final RemoteEJBReceiver ejbReceiver = context.getClientContext().getAttachment(RemoteTransportProvider.ATTACHMENT_KEY);
-            if (ejbReceiver != null) {
-                final EJBClientChannel ejbClientChannel = context.getAttachment(RemoteEJBReceiver.EJBCC_KEY);
-                if (ejbClientChannel != null) {
-                    final NodeInformation nodeInformation = ejbReceiver.getDiscoveredNodeRegistry().getNodeInformation(((NodeAffinity) targetAffinity).getNodeName());
-                    if (nodeInformation != null) {
-                        nodeInformation.removeModule(ejbClientChannel, context.getLocator().getIdentifier().getModuleIdentifier());
+            final EJBLocator<?> locator = context.getLocator();
+            if (locator.isStateful()) {
```

Comment:
@chengfang I think it is indeed a good suggestion: add a comment here with a link to the Jira so we can refer to it in the future in case we have quesitons about why is this check needed.

Reply:
@fl4via I added comments and updated the PR.  @rachmatowicz thanks for the detailed review and analysis.